common.sh

```diff
@@ -187,9 +187,39 @@ create_instance()
 get_instance_ip()
 {
     execution_seq=$((${execution_seq}+1))
-    INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
-                        --query "Reservations[*].Instances[*].PrivateIpAddress" \
-                        --output=text)
+    local retry=20
+    local ret=0
+    local bash_option=$-
+    local restore_e=0
+    local get_instance_ip_succeed=0
+    local instance_ips_array=()
+    if [[ $bash_option =~ e ]]; then
+        restore_e=1
+        set +e
+    fi
+    while [ $retry -ge 0 ]; do
+        INSTANCE_IPS=$(aws ec2 describe-instances --instance-ids ${INSTANCE_IDS[@]} \
+                            --query "Reservations[*].Instances[*].PrivateIpAddress" \
+                            --output=text)
+        ret=$?
+        instance_ips_array=($INSTANCE_IPS)
+        if [ "$ret" -eq 0 ] && [ -n "$INSTANCE_IPS" ] && [ "${#instance_ips_array[@]}" == "${#INSTANCE_IDS[@]}" ]; then
+            echo "Successfully get instance ips: ${INSTANCE_IPS}."
+            get_instance_ip_succeed=1
+            break
+        else
+            echo "Failed to get instance ips, retry after 10 seconds."
+            sleep 10
+        fi
+        retry=$((retry-1))
+    done
+    if [ "$get_instance_ip_succeed" -eq 0 ]; then
```

Comment:
if you're using -eq, there's no need for quotes.  You'd either get a parse error or a test failure if the variable expanded to the empty string.  You don't use quotes below, you should be consistent here.

Reply:
Thanks, will fix.

Reply:
Fixed in latest revision.